src/executor/stack/executor.rs

```diff
@@ -839,18 +847,10 @@ impl<'config, 'precompiles, S: StackState<'config>, P: PrecompileSet>
 
 		self.enter_substate(gas_limit, false);
 
-		{
-			if self.code_size(address) != U256::zero() {
-				let _ = self.exit_substate(StackExitKind::Failed);
-				return Capture::Exit((ExitError::CreateCollision.into(), None, Vec::new()));
-			}
-
-			if self.nonce(address) > U256::zero() {
-				let _ = self.exit_substate(StackExitKind::Failed);
-				return Capture::Exit((ExitError::CreateCollision.into(), None, Vec::new()));
-			}
-
-			self.state.reset_storage(address);
+		// Check create collision: EIP-7610
+		if self.is_create_collision(address) {
+			let _ = self.exit_substate(StackExitKind::Failed);
+			return Capture::Exit((ExitError::CreateCollision.into(), None, Vec::new()));
```

Comment:
Did you miss `self.state.reset_storage(address);` or it's not needed anymore?

Reply:
It checks storage is empty. We shouldn't touch storage data again, as it's empty anyway.